### ISSUE TITLE
tf.control_dependencies missing in dcgan-svhn/DCGAN.ipynb

### ISSUE THREAD

**Issue by angelmtenor** [udacity/deep-learning#128]
Since we are using batch normalization, tf.control_dependencies should be used here:
    with tf.control_dependencies(tf.get_collection(tf.GraphKeys.UPDATE_OPS)):  
        d_train_opt = tf.train.AdamOptimizer(learning_rate, beta1=beta1).minimize(d_loss, var_list=d_vars)
        g_train_opt = tf.train.AdamOptimizer(learning_rate, beta1=beta1).minimize(g_loss, var_list=g_vars)

Source:
https://github.com/udacity/deep-learning/blob/master/batch-norm/Batch_Normalization_Lesson.ipynb

**Comment by clapollo** [comment]
Yeah, this project definitely doesn't do batch normalization correctly. It's also never setting the training field properly. I'll look into it.

**Comment by clapollo** [comment]
Ok, I added the missing bits and changed a few comments to point out the important stuff.
